samplers/aws/xray/client.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type xrayClient struct {
+	// http client for sending unsigned proxied requests to the collector
+	httpClient *http.Client
+
+	endpoint string
```

Comment:
There is a lot of manual URL manipulation in this code. Using a dedicated and tested type here will help prevent bugs and provide needed untilities.

```suggestion
	endpoint *url.URL
```